api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1859,6 +1859,97 @@ type AzurePlatformSpec struct {
 	// +immutable
 	// +required
 	SecurityGroupID string `json:"securityGroupID,omitempty"`
+
+	// ManagedIdentities contains the client IDs related to the managed identities needed for HCP control plane
+	// and data plane components that authenticate with Azure's API.
+	//
+	// +kubebuilder:validation:Required
```

Comment:
The AzurePlatformSpec has recently been added as it not currently shipped right? Adding new required fields to something that already has shipped would cause issues

Reply:
It's not been shipped for customers. This is part of ARO HCP so these fields will only be used by them. This PR will need to be backported to 4.17.z though as that is what they will be using to private preview.